extensions/widgets/segment_new/segment_new.lcb

```diff
@@ -0,0 +1,716 @@
+	put pProperties["selectedSegment"]	into mSelectedSegments
+	
+end handler
+----------
+
+----------
+-- this handler is called when the widget is created
+public handler OnCreate()
+	
+	-- properties
+	put "ios-8" into mTheme
+	put "segment" into mStyle
+	put true into mMultiSelect
+	
+	put ["left","centre","right"] into mSegmentNames
+	put ["Left","Centre","Right"] into mSegmentLabels
+	put ["0xf036","0xf037","0xf038"] into mSegmentIcons
+	put ["0xf036","0xf037","0xf038"] into mSelectedIcons
+	
+	put "icon" into mSegmentDisplay
+	put [30,30,30] into mSegmentMinWidth
+	put [1] into mSelectedSegments
+	--
+	
+	-- variables
+	put true into mGeometryIsChanged
+	put (the number of elements in mSegmentNames) into mNumSegments
+	put 0 into mLeft
+	put the empty path into mPerimeter
+	put the empty list into mLines
+	put 0 into mRadius
+	--
+	
+end handler
+----------
+
+----------
+-- this handler is called whenever LiveCode needs to redraw the widget
+public handler OnPaint()
+	
+	if mGeometryIsChanged then
+		-- update mPerimeter and mLines variables if the geometry has changed
+		put (the trunc of my height)/5 into mRadius
+		updatePerimeter()
+		updateLines()
+	end if
+	
+	drawSegments()
+	
+	-- draw the lines to separate the segments
+	set the antialias of this canvas to false
+	set the stroke width of this canvas to 1
+	set the paint of this canvas to fetchPaint("lines")
+	
+	variable tLine
+	repeat for each element tLine in mLines
+		stroke tLine on this canvas
+	end repeat
+	
+	set the antialias of this canvas to true
+	--
+	
+	-- draw the perimeter of the control
+	set the paint of this canvas to fetchPaint("lines")
+	set the stroke width of this canvas to 1
+	--stroke rectangle path of mPerimeter on this canvas
+	stroke mPerimeter on this canvas
+	--
+	
+	put false into mGeometryIsChanged
+end handler
+----------
+
+----------
+-- this handler is called whenever the geometry is changed
+public handler OnGeometryChanged()
+	put true into mGeometryIsChanged
+end handler
+----------
+
+----------
+-- this handler is called when the mouse is clicked
+public handler OnClick(in pXLoc as integer, in pYLoc as integer, in pMouseButton as integer, in pClicks as integer)
+	
+	variable tX as integer
+	variable tLabel as string
+	variable tSegment
+	variable tCount as integer
+	 
+	if mMultiSelect is false then
+		-- if cannot multiselect, then need to select the clicked-on segment and deselect the currently selected segment
+		-- if the clicked-on segment is already selected, then do nothing
+	
+		if the click position is not within fetchBounds(element 1 of mSelectedSegments) then
+			repeat with tX from 1 up to mNumSegments
+			
+				if the click position is within fetchBounds(tX) then
+					
+					put the empty list into mSelectedSegments
+					push tX onto back of mSelectedSegments
+					
+					put element tX of mSegmentNames into tLabel
+					dispatch command "segmentSelected" to my script object with [tLabel]
```

Comment:
This needs to change to use the new 'post' syntax so that it is safe from re-entrancy issues.